### DISCUSSION TITLE
Can I get images from AWS s3 by using Coil and AWS SDK for Kotlin

### DISCUSSION THREAD

**Discussion by GarikKhachaturyan** [aws/aws-sdk-kotlin#987] [Q&A]
I know that I can use AWS API for that and make authentication in the Coil client interceptor, but is there a way to redirect network calls to [AWS SDK for Kotlin](https://github.com/awslabs/aws-sdk-kotlin) or somehow authenticate Coil's request by AWS SDK because the AWS SDK already handles request authentication?
In this case, the main reason that I want to use Coil is that it will handle disk and memory caching.

**Comment by aajtodd** [comment]
Thanks for the discussion but this seems like more of a question for the Coil folks about how to best integrate S3 and Coil. I suggest reaching out to them for guidance.

**Reply by GarikKhachaturyan** [reply]
Hi @aajtodd. Actually, I wrote them first but they haven't responded )). 
I saw that AWS SDK also uses OkHttpClient and you can pass that client to Coil but the question is how can I get preconfigured OkHttpClient instance from AWS SDK?

**Reply by aajtodd** [reply]
The HTTP client used by the AWS SDK doesn't handle authentication or anything other than sending and receiving of HTTP requests and responses. By default the SDK uses OkHttp but it can be configured to a different HTTP client (you hand each AWS service client, e.g. S3, an HTTP client instance to use). This is really a question for Coil since it isn't about how to use the AWS SDK for Kotlin.

**Reply by GarikKhachaturyan** [reply]
Okay. Anyway, thanks for your response.
